src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100, hidden = false) {
+    this.name = name;
+    this.health = health;
+    this.hidden = hidden;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    const animalIndex = Animal.alive.indexOf(target);
+
+    if (target.hidden !== true && target instanceof Herbivore) {
+      target.health -= 50;
+    }
+
+    if (target.health <= 0) {
```

Comment:
The logic to remove the dead animal from 'Animal.alive' array should be placed outside of the 'bite' method to avoid repeated code and to handle the case when an animal's health becomes 0 outside of a 'bite' context. Consider creating a separate method to manage the animal's life state.